src/autoscaler/healthendpoint/health_readiness_test.go

```diff
@@ -318,4 +322,55 @@ var _ = Describe("Health Readiness", func() {
 
 	})
 
+	Context("pprof endpoint", func() {
+		When("basic auth is not configured", func() {
+			BeforeEach(func() {
+				config.HealthCheckUsername = ""
+				config.HealthCheckPassword = ""
+			})
+			It("should not be available", func() {
+				apitest.New().
+					Handler(healthRoute).
+					Get("/debug/pprof").
+					Expect(t).
+					Assert(bodyContainsSubstring(false, "Types of profiles available")).
+					Status(http.StatusOK).
+					End()
+			})
+		})
+
+		When("basic auth is configured", func() {
+			It("should be available", func() {
```

Comment:
how about a test without the credentials ?

Reply:
yep, missing.

Reply:
added now